apps/miiverse-api/src/middleware/auth.ts

```diff
@@ -41,52 +40,50 @@ async function auth(request: express.Request, response: express.Response, next:
 	}
 
 	if (!encryptedToken) {
-		return badAuth(response, 15, 'NO_TOKEN');
+		return badRequest(response, ApiErrorCode.NO_TOKEN);
 	}
 
 	const pid: number = getPIDFromServiceToken(encryptedToken);
 	if (pid === 0) {
-		return badAuth(response, 16, 'BAD_TOKEN');
+		return badRequest(response, ApiErrorCode.BAD_TOKEN);
 	}
 
 	const paramPack = getValueFromHeaders(request.headers, 'x-nintendo-parampack');
 	if (!paramPack) {
-		return badAuth(response, 17, 'NO_PARAM');
+		return badRequest(response, ApiErrorCode.NO_PARAM_PACK);
 	}
 
 	const paramPackData = decodeParamPack(paramPack);
 	const paramPackCheck = ParamPackSchema.safeParse(paramPackData);
 	if (!paramPackCheck.success) {
-		console.log(paramPackCheck.error);
-		return badAuth(response, 18, 'BAD_PARAM');
+		request.log.error(paramPackCheck.error, 'Failed to parse param pack');
+		return badRequest(response, ApiErrorCode.BAD_PARAM_PACK);
 	}
 
 	let user: GetUserDataResponse;
 
 	try {
 		user = await getUserAccountData(pid);
 	} catch (error) {
-		// TODO - Log this error
-		console.log(error);
-		return badAuth(response, 18, 'BAD_PARAM');
+		request.log.error(error, `Failed to get account data for ${pid}`);
+		return serverError(response, ApiErrorCode.ACCOUNT_SERVER_ERROR);
 	}
 
-	let discovery: HydratedEndpointDocument | null;
-
-	if (user) {
-		discovery = await getEndpoint(user.serverAccessLevel);
-	} else {
-		discovery = await getEndpoint('prod');
-	}
+	const discovery = await getEndpoint(user.serverAccessLevel);
 
 	if (!discovery) {
-		return badAuth(response, 19, 'NO_DISCOVERY');
+		request.log.error(user, `Discovery data is missing for ${user.serverAccessLevel}`);
+		return serverError(response, ApiErrorCode.NO_DISCOVERY_DATA);
 	}
 
-	if (discovery.status !== 0) {
-		return serverError(response, discovery);
+	if (discovery.status > 0 && discovery.status <= 7) {
+		return badRequest(response, discovery.status);
+	} else if (discovery.status !== 0) {
+		request.log.error(discovery, `Discovery for ${user.serverAccessLevel} has unexpected status`);
```

Comment:
I think it's better if we don't log normal objects as if they are errors. Instead we can add the invalid param to the log message:
```ts
request.log.error(`Discovery status ${discovery.status} for ${user.serverAccessLevel} is unexpected`)
```